lib/features/steps/request_assertion_steps.rb

```diff
@@ -137,6 +137,14 @@ def find_request(request_index)
   assert_not_nil(parsed_time)
 end
 
+Then(/^the payload has a valid sessions array(?: for request (\d+))?$/) do |request_index|
+  sessions = read_key_path(find_request(request_index)[:body], "sessions")
+  sessionCounts = read_key_path(find_request(request_index)[:body], "sessionCounts")
+  value = sessions || sessionCounts
+  assert_kind_of Array, value
+  assert(value.length > 0, "the payload must contain a non empty sessions or sessionCounts array")
+end
+
```

Comment:
minor nitpick, `find_request(request_index)[:body]` could be extracted to a variable here

Reply:
Should this be in `session_tracking_steps.rb`? I suppose the argument could be made to store it in either file.